### ISSUE TITLE
error: Cannot find module './yoga.wasm' from '/home/okineadev/okineadev-npm/dist/cli.js'

### ISSUE THREAD

**Issue by okineadev** [oven-sh/bun#15639]
### What version of Bun is running?

1.1.39-canary.12+da5d4d791

### What platform is your computer?

Linux 5.15.167.4-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

1. Run `bun run build` (runs `bun build ./src/index.tsx --minify --outfile ./dist/cli.js --target node --footer '// Built with Bun! (https: //bun.sh)'`)
2. `bun dist/cli.js`
3. See this error:

```
error: Cannot find module './yoga.wasm' from '/home/okineadev/okineadev-npm/dist/cli.js'

Bun v1.1.39-canary.12+da5d4d791 (Linux x64)
```

### What is the expected behavior?

Bun will correctly compile all files and dependencies including the `yoga.wasm` file that is required for the `yoga-wasm-web` package to run, and then the compiled file will run successfully without any errors

### What do you see instead?

```
error: Cannot find module './yoga.wasm' from '/home/okineadev/okineadev-npm/dist/cli.js'
```

When i run `bun dist/cli.js`

### Additional information

I found the following lines in the `yoga-wasm-web` package files in `node_modules`:

- `node_modules/yoga-wasm-web/dist/browser.js`:

  ```js
  let Yoga=await E(await fetch(new URL("./yoga.wasm",import.meta.url)).then(E=>E.arrayBuffer()));export{Yoga as default};
  ```

- `node_modules/yoga-wasm-web/dist/node.js`

  ```js
  let Yoga=await a(await E(_(import.meta.url).resolve("./yoga.wasm")));export{Yoga as default};
  ```

As I understand it, Bun does not correctly resolve / see the `./yoga.wasm` file
The file itself is present in the `node_modules/yoga-wasm-web/dist` directory

### Alternatives

To solve this problem, I modified the build script to manually move the `yoga.wasm` file to the `dist` folder:

```json
"scripts": {
  "build": "bun build ./src/index.tsx ... && cp node_modules/yoga-wasm-web/dist/yoga.wasm dist/"
}
```

This is a temporary crutch 🩼

**Comment by RiskyMH** [comment]
Yeah this is a little interesting as it kinda doesn't explicitly say that it should be bundled.

So you could `--external=yoga-wasm-web` and then it will still be imported from node_modules (also if it does bundle, you would need to use `--outdir` not `--outfile`)

https://github.com/shuding/yoga-wasm-web/blob/main/node.js#L6

(your `ink` npm package imports via `import Yoga from 'yoga-wasm-web/auto';` which makes everything harder)

**Comment by okineadev** [comment]
> Yeah this is a little interesting as it kinda doesn't explicitly say that it should be bundled.
> 
> So you could `--external=yoga-wasm-web` and then it will still be imported from node_modules (also if it does bundle, you would need to use `--outdir` not `--outfile`)
> 
> https://github.com/shuding/yoga-wasm-web/blob/main/node.js#L6
> 
> (your `ink` npm package imports via `import Yoga from 'yoga-wasm-web/auto';` which makes everything harder)

Thanks, I'll check it out tomorrow

**Comment by okineadev** [comment]
> Yeah this is a little interesting as it kinda doesn't explicitly say that it should be bundled.
> 
> So you could `--external=yoga-wasm-web` and then it will still be imported from node_modules (also if it does bundle, you would need to use `--outdir` not `--outfile`)
> 
> [shuding/yoga-wasm-web@`main`/node.js#L6](https://github.com/shuding/yoga-wasm-web/blob/main/node.js#L6)
> 
> (your `ink` npm package imports via `import Yoga from 'yoga-wasm-web/auto';` which makes everything harder)

Yes, that helped, thanks!

**Comment by Plopix** [comment]
but it does not really work right, as soon as you move the binary it stops working, right @okineadev ?

**Comment by okineadev** [comment]
> but it does not really work right, as soon as you move the binary it stops working, right [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/81070564?s=64&amp;u=6334d6f9c37e20fbc90594191279f327739919cc&amp;v=4">@okineadev](https://github.com/okineadev?rgh-link-date=2025-01-11T02%3A44%3A42.000Z) ?

I don't know

**Comment by dotneet** [comment]
Note: This issue has been resolved in ink version 5.2.0. The yoga-wasm-web dependency has been replaced with yoga-layout.
